File: frontend/src/types/asset.ts
export type AssetQualityStatus = "approved" | "draft" | "rejected";

export type Asset = {
  id: string;
  name: string;
  category: string;
  tags: string[];
  style: string[];
  colors: string[];
  file: string;
  file_url: string;
  license: string;
  source: string;
  quality_status: AssetQualityStatus;
};

File: backend/app/schemas/asset.py
from pydantic import BaseModel


class AssetRead(BaseModel):
    id: str
    name: str
    category: str
    tags: list[str]
    style: list[str]
    colors: list[str]
    file: str
    file_url: str
    license: str
    source: str
    quality_status: str
